rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_equipment_type.py

```diff
@@ -0,0 +1,107 @@
+    SERVICE_WATER_HEATER_TANK.COMMERCIAL_STORAGE,
+]
+
+
+class GetSWHEquipmentType:
+    ELECTRIC_RESISTANCE_INSTANTANEOUS = "ELECTRIC_RESISTANCE_INSTANTANEOUS"
+    ELECTRIC_RESISTANCE_STORAGE = "ELECTRIC_RESISTANCE_STORAGE"
+    ELECTRIC_RESISTANCE_OTHER = "ELECTRIC_RESISTANCE_OTHER"
+    GAS_INSTANTANEOUS = "GAS_INSTANTANEOUS"
+    GAS_STORAGE = "GAS_STORAGE"
+    GAS_OTHER = "GAS_OTHER"
+    OIL_INSTANTANEOUS = "OIL_INSTANTANEOUS"
+    OIL_STORAGE = "OIL_STORAGE"
+    OIL_OTHER = "OIL_OTHER"
+
+
+def get_swh_equipment_type(rmd: dict, service_water_heating_equipment_id: str) -> str:
+    """
+    This function determines whether the swh equipment type is one of: (ELECTRIC_RESISTANCE_INSTANTANEOUS, ELECTRIC_RESISTANCE_STORAGE, GAS_STORAGE, OTHER)
+
+    Parameters
+    ----------
+    rmd: dict
+        RMD at RuleSetModelDescription level
+    service_water_heating_equipment_id: str
+        service water heating equipment id
+
+    Returns
+    -------
+    type: string
+        ex: type = "ELECTRIC_RESISTANCE_INSTANTANEOUS"
+    """
+
+    service_water_heating_equipment = find_exactly_one_with_field_value(
+        "$.service_water_heating_equipment[*]",
+        "id",
+        service_water_heating_equipment_id,
+        rmd,
+    )
+
+    swh_tank_type = getattr_(
+        service_water_heating_equipment,
+        "service_water_heating_equipment",
+        "tank",
+        "type",
+    )
+    swh_type = getattr_(
+        service_water_heating_equipment,
+        "service_water_heating_equipment",
+        "heater_type",
+    )
+    fuel_type = getattr_(
+        service_water_heating_equipment,
+        "service_water_heating_equipment",
+        "heater_fuel_type",
+    )
+
+    assert_(
+        fuel_type
+        in [
+            ENERGY_SOURCE.ELECTRICITY,
+            ENERGY_SOURCE.NATURAL_GAS,
+            ENERGY_SOURCE.FUEL_OIL,
+        ],
+        "Fuel type must be one of `ELECTRICITY`, `NATURAL_GAS`, `FUEL_OIL`.",
+    )
+
+    if swh_type == SERVICE_WATER_HEATER.CONVENTIONAL:
+        if swh_tank_type in INSTANTANEOUS_TYPE:
+            if fuel_type == ENERGY_SOURCE.ELECTRICITY:
+                type = GetSWHEquipmentType.ELECTRIC_RESISTANCE_INSTANTANEOUS
+            elif fuel_type == ENERGY_SOURCE.NATURAL_GAS:
+                type = GetSWHEquipmentType.GAS_INSTANTANEOUS
+            elif fuel_type == ENERGY_SOURCE.FUEL_OIL:
+                type = GetSWHEquipmentType.OIL_INSTANTANEOUS
+
+        elif swh_tank_type in STORAGE_TYPE:
+            if fuel_type == ENERGY_SOURCE.ELECTRICITY:
+                type = GetSWHEquipmentType.ELECTRIC_RESISTANCE_STORAGE
+            elif fuel_type == ENERGY_SOURCE.NATURAL_GAS:
+                type = GetSWHEquipmentType.GAS_STORAGE
+            elif fuel_type == ENERGY_SOURCE.FUEL_OIL:
+                type = GetSWHEquipmentType.OIL_STORAGE
+    else:
+        if fuel_type == ENERGY_SOURCE.ELECTRICITY:
+            type = GetSWHEquipmentType.ELECTRIC_RESISTANCE_OTHER
```

Comment:
[https://github.com/pnnl/ruleset-checking-tool/blob/b35403785a307f96cea5b8a4cd2c73fb28d9cc17/docs/ruleset_functions/get_swh_equipment_type.md#:~:text=set%20the%20type%20to%20OTHER.%20If%20the%20swh_tank_type%20is%20an%20identifiable%20type%2C%20type%20will%20be%20overwritten%20to%20that%20type%3A%20type%20%3D%20%22OTHER%22](url)

Reply:
That is correct. We do need an `OTHER` type in the types.
So in here it is:
```python
else:
    type = GetSWHEquipmentType.OTHER
```

Reply:
Good point! Addressed.